Dockerfile

```diff
@@ -1,17 +1,29 @@
-FROM oamdev/docker-terraform-base:1.0.9
+FROM alpine:3.13
+
+RUN \
+  apk update && \
+  apk add bash py-pip && \
+  apk add --virtual=build gcc libffi-dev musl-dev openssl-dev python3-dev make && \
+  apk add curl jq python3 ca-certificates git openssl unzip wget && \
+  pip --no-cache-dir install -U pip && \
+  pip install azure-cli && \
+  apk del --purge build
 
 VOLUME ["/data"]
 
 WORKDIR /data
 
-ENTRYPOINT ["tail", "-f", "/dev/null"]
-
 ENV TERRAFORM_VERSION=1.0.2
 COPY terraform_${TERRAFORM_VERSION}_linux_amd64.zip /tmp
 RUN cd /tmp && \
     unzip terraform_${TERRAFORM_VERSION}_linux_amd64.zip -d /usr/bin
+COPY retrieve_tf_provider.sh /tmp
 
-ARG VCS_REF
+ENV RETRIEVE_TF_PROVIDER=/tmp/retrieve_tf_provider.sh
 
-LABEL org.label-schema.vcs-ref=$VCS_REF \
-    org.label-schema.vcs-url="https://github.com/broadinstitute/docker-terraform"
+RUN $RETRIEVE_TF_PROVIDER random 3.1.0
+RUN $RETRIEVE_TF_PROVIDER alicloud 1.140.0
```

Comment:
Maybe always use the newest version is better.

Reply:
alicloud or random?